### PR TITLE
Fixes rainbow crayons and spraycans not drawing where you click

### LINKED ISSUE #43143: Rainbow Crayon Doesn't Write In The Spot You Clicked
Issue reported from  Round ID: 103901 (/tg/Station Sybil [ENGLISH] [US-EAST] [100% LAG FREE])

Reporting client version: 512

image

Title. Trying to write a straight sentence makes it all... weird, like the 'captain' there, almost unreadable.

Steps:
1) Spawn as clown or whatever.
2) Open the crayon menu.
3) Click 'new text' at the bottom of the menu. Enter whatever you want.
4) Write a bunch of stuff on the floor; some of the writing should be askew.

### PR DESCRIPTION
**Pull request by ShizCalev**
Fixes #43143
:cl: ShizCalev
fix: Cleaned up some crayon and spraycan code for futureproofing.
/:cl:

### CODE REVIEW

File: code/game/objects/items/crayons.dm [hunk 1]
```diff
@@ -488,7 +488,7 @@
 
 	charges = -1
 
-/obj/item/toy/crayon/rainbow/afterattack(atom/target, mob/user, proximity)
+/obj/item/toy/crayon/rainbow/afterattack(atom/target, mob/user, proximity, params)
```
Comment (MrDoomBringer) [inline code/game/objects/items/crayons.dm]: what does adding that parameter do?

Comment (ShizCalev) [reply code/game/objects/items/crayons.dm]: https://github.com/tgstation/tgstation/blob/227983a777c0c76be3a96266bf5679aa6b12524e/code/game/objects/items/crayons.dm#L343-L349

passes the click location

Comment (MrDoomBringer) [reply code/game/objects/items/crayons.dm]: oh, neat. ty

**Comment by SpaceManiac** [comment]
Unable to reproduce this bug in the first place, not convinced this change would fix it even if I could

**Comment by ShizCalev** [comment]
yeah he must've been drunk. did this before even testing it. 

oh well, code improvement i guess.

**Review by SpaceManiac — APPROVED** [review]

**Pull request merged**